packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -8,32 +8,51 @@ import 'dart:ui' as ui show Codec, FrameInfo, Image;
 import 'package:flutter/foundation.dart';
 import 'package:flutter/scheduler.dart';
 
-const String _flutterWidgetsLibrary = 'package:flutter/widgets.dart';
+const String _flutterPaintingLibrary = 'package:flutter/painting.dart';
 
 /// A [dart:ui.Image] object with its corresponding scale.
 ///
 /// ImageInfo objects are used by [ImageStream] objects to represent the
 /// actual data of the image once it has been obtained.
 ///
-/// The receiver of an [ImageInfo] object must call [dispose]. To safely share
-/// the object with other clients, use the [clone] method before calling
-/// dispose.
+/// The receiver of an [ImageInfo] object must call [dispose].
 @immutable
 class ImageInfo {
   /// Creates an [ImageInfo] object for the given [image] and [scale].
   ///
+  /// The created [ImageInfo] will dispose the [image] when [dispose] is called.
+  /// If the [image] needs to be shared with other clients, pass `image.clone()`
+  /// The cloning will not increase memory footprint, because it clones
+  /// just handler to the image.
+  ///
+  /// After passing [image] to the constructor,
+  /// remove any references to it so that the image can be garbage collected,
+  /// after ImageInfo disposal.
```

Comment:
I'm not sure I understand what this means. The image is non-nullable - how do I remove references? 

I think I know what this means, but maybe an example would be helpful.